src/Core/src/Platform/Windows/NavigationManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Media.Animation;
+using Microsoft.UI.Xaml.Navigation;
+
+namespace Microsoft.Maui
+{
+	public class NavigationManager
+	{
+		IView? _currentPage;
+		IMauiContext _mauiContext;
+		NavigationFrame? _navigationFrame;
+		protected WindowManager WindowManager => _mauiContext.GetWindowManager();
+		private protected INavigationView? NavigationView { get; private set; }
+		public IReadOnlyList<IView> NavigationStack { get; set; } = new List<IView>();
+		public IMauiContext MauiContext => _mauiContext;
+		public IView CurrentPage
+			=> _currentPage ?? throw new InvalidOperationException("CurrentPage cannot be null");
+		public NavigationFrame NavigationFrame =>
+			_navigationFrame ?? throw new InvalidOperationException("NavigationFrame Null");
+
+		public NavigationManager(IMauiContext mauiContext)
+		{
+			_mauiContext = mauiContext;
+		}
+
+		public virtual void Connect(IView navigationView, NavigationFrame navigationFrame)
+		{
+			if (_navigationFrame != null)
+				_navigationFrame.Navigated -= OnNavigated;
+
+			navigationFrame.Navigated += OnNavigated;
+			_navigationFrame = navigationFrame;
+			NavigationView = (INavigationView)navigationView;
+		}
+
+		public virtual void NavigateTo(NavigationRequest args)
+		{
+			bool push = true;
+			if (NavigationStack.Count > args.NavigationStack.Count)
+				push = false;
+
+			IReadOnlyList<IView> newPageStack = args.NavigationStack;
+			bool animated = args.Animated;
+			var previousNavigationStack = NavigationStack;
+			var previousNavigationStackCount = previousNavigationStack.Count;
+			bool initialNavigation = NavigationStack.Count == 0;
+
+			// User has modified navigation stack but not the currently visible page
+			// So we just sync the elements in the stack
+			if (!initialNavigation &&
+				newPageStack[newPageStack.Count - 1] == 
+				previousNavigationStack[previousNavigationStackCount - 1])
+			{
+				SyncBackStackToNavigationStack(newPageStack);
+				NavigationStack = new List<IView>(newPageStack);
+				NavigationView?.NavigationFinished(NavigationStack);
+				return;
+			}
+
+
+			NavigationTransitionInfo? transition = null;
+
+			if (animated)
+			{
+				transition = new SlideNavigationTransitionInfo()
```

Comment:
Can we make this something like 

`public virtual NavigationTransitionInfo GetNavigationTransition()`

Reply:
👍

Reply:
The animations only ever animate with a slide left animation :-/ 

I think we need to switch to Connected Animations
https://github.com/dotnet/maui/issues/2604